custom_model_runner/datarobot_drum/runtime_parameters/runtime_parameters.py

```diff
@@ -109,57 +101,80 @@ class RuntimeParametersLoader:
+        if not code_dir:
+            raise InvalidInputFilePath("Empty code-dir path!")
+
+        self._load_parameter_definitions(code_dir)
 
         try:
             with open(values_filepath, encoding="utf-8") as file:
-                try:
-                    self._yaml_content = yaml.safe_load(file)
-                    if not self._yaml_content:
-                        raise InvalidEmptyYamlContent(
-                            "Runtime parameter values YAML file is empty!"
-                        )
-                except yaml.YAMLError as exc:
-                    raise InvalidYamlContent(
-                        f"Invalid runtime parameter values YAML content! {str(exc)}"
-                    )
+                self._yaml_content = yaml.safe_load(file)
+            if not self._yaml_content:
+                raise InvalidEmptyYamlContent("Runtime parameter values YAML file is empty!")
+        except yaml.YAMLError as exc:
+            raise InvalidYamlContent(f"Invalid runtime parameter values YAML content! {str(exc)}")
         except FileNotFoundError:
             raise InvalidInputFilePath(
                 f"Runtime parameter values file does not exist! filepath: {values_filepath}"
             )
 
+    def _load_parameter_definitions(self, code_dir):
+        try:
+            with open(os.path.join(code_dir, MODEL_CONFIG_FILENAME)) as file:
+                model_metadata = yaml.safe_load(file)
+            if not model_metadata:
+                raise InvalidEmptyYamlContent("Model-metadata YAML file is empty!")
+        except yaml.YAMLError as exc:
+            raise InvalidYamlContent(f"Invalid model-metadata YAML content! {str(exc)}")
+        except FileNotFoundError:
+            raise InvalidInputFilePath(
+                f"{MODEL_CONFIG_FILENAME} must exist to use runtime parameters"
+            )
+
+        parameters = model_metadata.get(ModelMetadataKeys.RUNTIME_PARAMETERS)
+        if not parameters:
+            raise InvalidYamlContent(
+                f"{MODEL_CONFIG_FILENAME}: YAML file must contain at least one "
+                f"parameter definition in the section '{ModelMetadataKeys.RUNTIME_PARAMETERS}'"
+            )
+        self._parameter_definitions = {}
+        for parameter in parameters:
+            try:
+                data = RuntimeParameterDefinitionTrafaret.check(parameter)
+            except t.DataError as exc:
+                raise ErrorLoadingRuntimeParameter(f"Failed to load runtime parameter: {str(exc)}")
+            self._parameter_definitions[data["name"]] = self.ParameterDefinition(**data)
+
     def setup_environment_variables(self):
         credential_payload_trafaret = RuntimeParameterCredentialPayloadTrafaret()
         string_payload_trafaret = RuntimeParameterStringPayloadTrafaret()
-        for param_key, param_value in self._yaml_content.items():
+        for param_key, param_definition in self._parameter_definitions.items():
```

Comment:
so this means that all parameter definitions always have a default value, correct?  is it possible for the user's model metadata to have no default set?  if so, is it None?

Reply:
Yup, the default _default_ 😉 is `None`

Reply:
:scream: